elastic/kibana
# [Response Ops][Alerting] Add handling for space-specific resource installation in FAAD

Resolves https://github.com/elastic/kibana/issues/153115

## Summary

* Adds `isSpaceAware` flag to alerting FAAD registration. When this is set to true, namespace-specific resources will be installed ad-hoc when rules are created in custom spaces
* Inside the alerting task runner, we check and wait for FAAD resource installation promises. This ensures that any FAAD resource (component templates, index templates, indices) are done installing at the beginning of rule execution (before any rule type executors or lifecycle/persistence executors are run).

## To Verify
1. Run this branch with `enableFrameworkAlerts: true`. 
  a. Verify that we can create and preview detection rules in both the default space and a custom space. Verify that space-specific resources are created for the rule in a custom space (i.e. a `.internal.alerts-security.alerts-{spaceId}-000001` index)
  b. Verify that we can create an observability rule in both the default space and a custom space. Verify that no space specific resources are created and alerts for all obs rules are written to the default index.
3. Verify that when running this branch on rule registry rules (in default & custom spaces) from `main` or a previous version, the rules continue to run successfully.
